### PR TITLE
Fix: Updated version dependency pinning for typing annotation

### PR DESCRIPTION
**Pull request by tanzim-bgl**
Any related Github Issues?: _add link here_ 

_What has changed? This should match `CHANGELOG.md`._

# Check list

## Before asking for a review

- [x] Target branch is `master`
- [x] `make test` passes locally.
- [x] [`[Unreleased]`](../blob/master/CHANGELOG.md#unreleased) section in [CHANGELOG.md](../blob/master/CHANGELOG.md) is updated.
- [x] I reviewed the "Files changed" tab and self-annotated anything unexpected.

Updates have been made to unreleased section in changelog.
Locally tested. 

## Before review (reviewer)

- [x] All of the above are checked and true. **Review ALL items.** If not, return the PR to the author.
- [ ] Continuous Integration is passing. If not, return the PR to the author.

## After merge (reviewer)

- [ ] Any issues that may now be closed are closed.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@ PyYAML~=5.4
 python-dateutil~=2.8
 jsonpointer~=2.0
 json-ref-dict>=0.6.2,<0.8.0
-typing_extensions~=3.7
+typing_extensions~=4.2.0
```
Comment (jacksmith15) [inline requirements.txt:5]: I think we can keep this very loose, so it still works with packages which depend on `typing_extensions<4.0.0`:

```suggestion
typing_extensions>=3.7.0,<5.0.0
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,7 +14,11 @@ Types of changes are:
 
 ## [Unreleased]
 ### Changed
-* Typo in quickstart.rst
+* Relaxed dependency version pinning for typing-annotations
+
+## [0.13.6] - 2023-02-16
+### Fixed
+* Relazed dependency version pinning for typing-annotations
```
Comment (jacksmith15) [inline CHANGELOG.md:21]:
```suggestion
```

The release tool will handle version bumping post-merge :+1:

File: statham/__init__.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "0.13.5"
+__version__ = "0.13.6"
```
Comment (jacksmith15) [inline statham/__init__.py:1]:
```suggestion
__version__ = "0.13.5"
```

As above, the release tool will handle version bumping post-merge

**Comment by jacksmith15** [comment]
Thanks for the PR!

I made some small suggestions. I'll cut a release immediately after merging.

File: requirements.txt (hunk 1 above)
Comment (tanzim-bgl) [reply requirements.txt:5]: This makes sense, thanks!

**Comment by tanzim-bgl** [comment]
Thank you for replying quickly to this! Thanks again for your awesome work on the library itself.

**Review by jacksmith15 — APPROVED** [review]

**Pull request merged**